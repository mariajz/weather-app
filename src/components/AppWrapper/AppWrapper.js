import React from 'react';
import { SafeAreaView, StatusBar, Platform } from 'react-native';

const AppWrapper = ({ children }) => {
    return (
        <>
            {Platform.OS === 'android' && (
                <StatusBar barStyle="light-content" backgroundColor="maroon" />
            )}
            <SafeAreaView>{children}</SafeAreaView>
        </>
    );
};

export default AppWrapper;
