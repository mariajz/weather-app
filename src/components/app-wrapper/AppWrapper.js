import { Provider } from 'jotai';
import React from 'react';
import { Platform, StatusBar } from 'react-native';
import { StyledSafeArea } from './AppWrapper.style';

const AppWrapper = ({ children }) => {
    return (
        <Provider>
            {Platform.OS === 'android' && (
                <StatusBar
                    barStyle="dark-content"
                    backgroundColor="transparent"
                />
            )}
            <StyledSafeArea>{children}</StyledSafeArea>
        </Provider>
    );
};

export default AppWrapper;
