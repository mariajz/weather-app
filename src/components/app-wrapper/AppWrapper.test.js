import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import AppWrapper from './AppWrapper';

jest.mock('../visual-elements', () => ({
    BackgroundImage: () => <MockView />,
}));

describe('AppWrapper', () => {
    it('should render SafeArea for android and ios', () => {
        const container = render(<AppWrapper />);

        expect(container).toMatchSnapshot();
    });
});
