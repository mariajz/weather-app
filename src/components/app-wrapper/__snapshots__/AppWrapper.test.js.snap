// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper should render SafeArea for android and ios 1`] = `
<RCTSafeAreaView
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
/>
`;
